control-plane/subcommand/server-acl-init/connect_inject.go

```diff
@@ -18,9 +18,7 @@ const defaultKubernetesHost = "https://kubernetes.default.svc"
 
 // configureConnectInject sets up auth methods so that connect injection will
 // work.
-func (c *Command) configureConnectInjectAuthMethod(consulClient *api.Client) error {
-
-	authMethodName := c.withPrefix("k8s-auth-method")
+func (c *Command) configureConnectInjectAuthMethod(consulClient *api.Client, authMethodName string) error {
```

Comment:
we can rename this function to a generic `configureAuthMethod` and also possibly pass in the bindingrule filters so it can be configurable for the connectInject case and also hard-coded for the component-auth-method case.